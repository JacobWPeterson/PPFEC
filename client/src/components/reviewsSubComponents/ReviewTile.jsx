/* eslint-disable camelcase */
import React from 'react';
import styled from 'styled-components';
import RatingStars from './RatingStars.jsx';

const TileContainer = styled.div`
  margin: 0 auto;
  padding: 10px 30px;
  border-bottom: 1px solid black;
`;

const monthNames = ['January', 'February', 'March', 'April', 'May', 'June',
  'July', 'August', 'September', 'October', 'November', 'December'];

const StarDateWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: baseline;
`;

const ResponseWrapper = styled.div`
  margin: 0 auto;
  backgound-color: lighblue;
`;

const ReviewThumbsWrapper = styled.img`
  display: inline-flex;
  justify-content: space-around;
  align-items: flex-star;
  border: 1px solid #ddd;
  padding: 5px;
  width: 80px;

  & :hover {
    box-shadow: 0 0 2px 1px rgba(0, 140, 186, 0.5);
  }
`;

class ReviewTile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    const { review } = this.props;
    const {
      date, rating, reviewer_name, summary, body, helpfulness,
    } = review;
    // Formatting date
    const reviewDate = new Date(date);
    const month = monthNames[reviewDate.getUTCMonth()];
    const day = reviewDate.getUTCDate();
    const year = reviewDate.getUTCFullYear();

    // calculating the percentage for stars
    const percentage = (rating / 5) * 100;

    // conditional rendering for recommend label
    let recommendLabel;
    let response;
    if (review.recommend) {
      recommendLabel = <div>âœ“ I recommend this product</div>;
    }
    // conditional rendering for response
    if (review.response) {
      response = (
        <ResponseWrapper>
          <h4>Response from seller</h4>
          <p>{review.response}</p>
        </ResponseWrapper>
      );
    }
    // conditional rendering for photos
    let photos;
    if (review.photos.length > 0) {
      photos = (
        <div>
          {review.photos.map((photo, index) => (
            <ReviewThumbsWrapper key={photo.id} src={photo.url} alt={`${index}reviewPhoto`} />
          ))}
        </div>
      );
    }

    return (
      <TileContainer>
        <StarDateWrapper>
          <RatingStars percent={percentage} />
          <span>
            { reviewer_name }
            ,
            {' '}
            {month}
            {' '}
            {day}
            ,
            {' '}
            {year}
          </span>
        </StarDateWrapper>
        <h3>{summary}</h3>
        <p>{body}</p>
        {photos}
        {recommendLabel}
        {response}
        <div>
          <span>
            Helpful?
            <a href="">Yes</a>
            (
            {helpfulness}
            )
          </span>
          {' '}
          |
          {' '}
          <span><a href="">Report</a></span>
        </div>
      </TileContainer>
    );
  }
}

export default ReviewTile;
