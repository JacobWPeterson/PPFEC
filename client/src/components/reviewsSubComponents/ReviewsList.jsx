import React from 'react';
import styled from 'styled-components';
import axios from 'axios';
import ReviewTile from './ReviewTile.jsx';
import Modal from './Modal.jsx';
import AddReviewForm from './AddReviewForm.jsx';

const ReviewsWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-basis: 65%;
  flex-wrap: nowrap;
  flex-direction: column;
  justify-content: center;
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-wrap: nowrap;
  flex-direction: row;
  align-items: baseline;
`;

const ListWrapper = styled.div`
  width: 100%;
  display: flex;
  max-height: 500px;
  overflow-y: scroll;
  flex-direction: column;
  ::-webkit-scrollbar {
    display: none;
  };
  -ms-overflow-style: none;
  scrollbar-width: none;
`;

const ReviewButton = styled.button`
  font-size: 15px;
  padding: 20px 10px;
  margin: 10px 15px;
  width: 200px;
  height: 55px;
`;

const ReviewSortWrapper = styled.div`
  display: flex;
  align-items: baseline;
  margin: 10px;
  padding: 5px;
`;

const SelectTag = styled.select`
  width: 120px;
  height: 100%;
  font-size: 20px;
  padding-left: 3px;
  text-decoration: underline;
  border: 0px;
  outline: 0px;
  font-weight: 600;
`;

class ReviewsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displayLimit: 2,
      reviewsArr: [],
      fullreviewsArr: [],
      tileMax: 0,
      addReviewShow: false,
      addReviewClick: false,
    };
    this.loadFirstTwoReviews = this.loadFirstTwoReviews.bind(this);
    this.loadMoreReviews = this.loadMoreReviews.bind(this);
    this.addReviewToggle = this.addReviewToggle.bind(this);
    this.sortSelected = this.sortSelected.bind(this);
    this.freshPage = this.freshPage.bind(this);
  }

  componentDidMount() {
    this.loadFirstTwoReviews(this.props.reviews.results);
  }

  loadFirstTwoReviews(data) {
    const displayArr = [];
    let tileCount = 0;
    while (tileCount < this.state.displayLimit) {
      displayArr.push(data[tileCount]);
      tileCount += 1;
    }
    this.setState({
      tileMax: data.length,
      reviewsArr: displayArr,
      fullreviewsArr: data,
    });
  }

  loadMoreReviews() {
    const loadArr = [];
    let count = 0;
    let totalLength = loadArr.length + this.state.reviewsArr.length;
    while (count < this.state.displayLimit && totalLength < this.state.tileMax) {
      loadArr.push(this.state.fullreviewsArr[totalLength]);
      count += 1;
      totalLength += 1;
    }
    this.setState((prevState) => ({
      reviewsArr: [...prevState.reviewsArr, ...loadArr],
    }));
  }

  addReviewToggle() {
    if (!this.state.addReviewShow) {
      this.setState({
        addReviewShow: true,
      });
    } else {
      this.setState({
        addReviewShow: false,
      });
    }
  }

  sortSelected(event) {
    const path = window.location.pathname;
    axios.get(`${path.slice(-6)}reviews/${event.target.value}`)
      .then((res) => {
        this.loadFirstTwoReviews(res.data.results);
        this.setState({
          fullreviewsArr: res.data.results,
        });
      })
      .catch((err) => { throw err; });

    event.preventDefault();
  }

  freshPage() {
    this.setState({
      addReviewClick: true,
    });
  }

  render() {
    // conditionlal rendering MORE VIEW button
    let moreReviewBtn;
    if (this.state.reviewsArr.length !== this.state.tileMax) {
      moreReviewBtn = <ReviewButton onClick={this.loadMoreReviews}>MORE REVIEWS</ReviewButton>;
    }

    const { reviews, reviewsMeta } = this.props;
    // get totalReviewCount
    let totalReviewCount = 0;
    Object.entries(reviewsMeta.ratings).forEach((rating) => {
      totalReviewCount += Number(rating[1]);
    });

    return (
      <ReviewsWrapper>
        <ReviewSortWrapper>
          <h2>
            {totalReviewCount}
            {' '}
            reviews, sorted by
          </h2>
          <SelectTag value={this.state.sortValue} onChange={this.sortSelected}>
            <option defaultValue="relevant">Relevant</option>
            <option value="helpful">Helpful</option>
            <option value="newest">Newest</option>
          </SelectTag>
        </ReviewSortWrapper>
        <ListWrapper>
          {this.state.reviewsArr.map(((review) => (
            <ReviewTile key={review.review_id} review={review} />)))}
        </ListWrapper>
        <ButtonWrapper>
          {moreReviewBtn}
          <ReviewButton onClick={this.addReviewToggle}>ADD A REVIEW   +</ReviewButton>
          {this.state.addReviewShow && (
          <Modal content={(
            <AddReviewForm
              toggle={this.addReviewToggle}
              productId={Number(reviews.product)}
              characteristics={reviewsMeta.characteristics}
              freshPage={this.freshPage}
            />
)}
          />
          )}
        </ButtonWrapper>
      </ReviewsWrapper>
    );
  }
}

export default ReviewsList;
