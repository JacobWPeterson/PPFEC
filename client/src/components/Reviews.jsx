import React, { useState } from 'react';
import styled from 'styled-components';
import axios from 'axios';
import ReviewsList from './reviewsSubComponents/ReviewsList.jsx';
import RatingSummary from './reviewsSubComponents/RatingSummary.jsx';
import RatingBreakDown from './reviewsSubComponents/RatingBreakDown.jsx';
import ProductBreakDown from './reviewsSubComponents/ProductBreakDown.jsx';

const ReviewsContainer = styled.div`
  width: 60vw;
  // height: 70vw;
  display: flex;
  flex-wrap: nowrap;
  flex-direction: column;
  align-items: flex-start;
  margin: 0 auto;
  padding: 10px 20px 10px 20px;
  font-family: 'Lato', sans-serif;
`;

const ReviewsTitle = styled.div`
  color: black;
  font-size: 20px;
  text-align: left;
`;

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  align-items: flex-start;
  justify-content: space-around;
`;

const RatingWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-basis: 35%;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  padding: 30px 30px;
`;

class Reviews extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: [],
      reviewsMeta: this.props.reviewsMeta,
      filterOn: false,
      product_id: this.props.product,
      entriesCount: this.props.count,
      page: this.props.page,
      displayLimit: 2,
      fullreviewsArr: [],
      sortSelection: 'relavent',
      addReviewShow: false,
    };
    this.starFilter = this.starFilter.bind(this);
    this.loadFirstTwoReviews = this.loadFirstTwoReviews.bind(this);
    this.sortSelected = this.sortSelected.bind(this);
    this.loadMoreReviews = this.loadMoreReviews.bind(this);
    this.addReviewToggle = this.addReviewToggle.bind(this);
  }

  componentDidMount() {
    this.loadFirstTwoReviews(this.props.reviews.results);
  }

  starFilter(star) {
    this.setState({
      fullreviewsArr: this.props.reviews.results.filter((review) => review.rating === star),
      filterOn: true,
    }, () => {
      this.loadFirstTwoReviews(this.state.fullreviewsArr);
    });
  }

  loadFirstTwoReviews(data) {
    const displayArr = [];
    let tileCount = 0;
    while (tileCount < this.state.displayLimit) {
      displayArr.push(data[tileCount]);
      tileCount += 1;
    }
    this.setState({
      reviews: displayArr,
      fullreviewsArr: data,
    });
  }

  loadMoreReviews() {
    const loadArr = [];
    let count = 0;
    let totalLength = loadArr.length + this.state.reviews.length;
    while (count < this.state.displayLimit && totalLength < this.state.fullreviewsArr.length) {
      loadArr.push(this.state.fullreviewsArr[totalLength]);
      count += 1;
      totalLength += 1;
    }
    this.setState((prevState) => ({
      reviews: [...prevState.reviews, ...loadArr],
    }));
  }

  sortSelected(event) {
    const path = window.location.pathname;
    axios.get(`${path.slice(-6)}reviews/${event.target.value}`)
      .then((res) => {
        this.loadFirstTwoReviews(res.data.results);
      })
      .catch((err) => { throw err; });

    event.preventDefault();
  }

  addReviewToggle() {
    if (!this.state.addReviewShow) {
      this.setState({
        addReviewShow: true,
      });
    } else {
      this.setState({
        addReviewShow: false,
      });
    }
  }

  render() {
    // console.log(this.state.reviews);
    // console.log(this.state.reviewsMeta);
    if (this.state.reviews.length > 0 && this.state.fullreviewsArr.length > 0) {
      return (
        <ReviewsContainer>
          <ReviewsTitle>RATINGS & REVIEWS</ReviewsTitle>
          <Wrapper>
            <RatingWrapper>
              <RatingSummary reviewsMeta={this.state.reviewsMeta} />
              <RatingBreakDown reviewsMeta={this.state.reviewsMeta} starFilter={this.starFilter} />
              <ProductBreakDown
                reviewsMeta={this.state.reviewsMeta}
              />
            </RatingWrapper>
            <ReviewsList
              loadFirstTwoReviews={this.loadFirstTwoReviews}
              fullreviewsArr={this.state.fullreviewsArr}
              reviews={this.state.reviews}
              reviewsMeta={this.state.reviewsMeta}
              loadMoreReviews={this.loadMoreReviews}
              sortSelected={this.sortSelected}
              addReviewToggle={this.addReviewToggle}
              addReviewShow={this.state.addReviewShow}
            />
          </Wrapper>
        </ReviewsContainer>
      );
    }
    return (
      <div>Loading Data</div>
    );
  }
}

export default Reviews;
