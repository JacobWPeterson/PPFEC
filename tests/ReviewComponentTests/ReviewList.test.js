/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import ReviewsList from '../../client/src/components/reviewsSubComponents/ReviewsList.jsx';
import ReviewTile from '../../client/src/components/reviewsSubComponents/ReviewTile.jsx';
import Reviews from '../../client/src/components/Reviews.jsx';

describe('ReviewList Component', () => {
  const props = {
    reviewsMeta: {
      product_id: '14930',
      ratings: { '1': '5', '2': '7', '3': '3', '4': '5', '5': '2' },
      recommended: { false: '5', true: '17' },
      characteristics: {
        Fit: { id: 50009, value: '3.0454545454545455' },
        Length: { id: 50010, value: '3.2727272727272727' },
        Comfort: { id: 50011, value: '2.8636363636363636' },
        Quality: { id: 50012, value: '3.1818181818181818' },
      },
    },
    fullreviewsArr: [
      {
        review_id: 147659,
        rating: 2,
        summary: 'Tenetur quo blanditiis necessitatibus eveniet quo ipsa quia alias voluptas.',
        recommend: true,
        response: null,
        body: 'Non amet dolorem ut tempora perferendis sint ut adipisci suscipit. Sint ea itaque debitis quos enim odit. Ad perferendis nihil quisquam est. Laboriosam fuga ea amet repellat. Blanditiis aspernatur nostrum aspernatur quas necessitatibus.',
        date: '2021-01-26T00:00:00.000Z',
        reviewer_name: 'Shyanne6',
        helpfulness: 12,
        photos: [],
      },
      {
        review_id: 147667,
        rating: 1,
        summary: 'Minima repudiandae eligendi nemo sit quia aspernatur.',
        recommend: false,
        response: null,
        body: 'Corporis voluptate dolore earum voluptas numquam. Perspiciatis cumque et. Dignissimos asperiores ab repudiandae. Consectetur et exercitationem hic. Ut repudiandae dolorum id.',
        date: '2021-01-20T00:00:00.000Z',
        reviewer_name: 'Janiya_Frami98',
        helpfulness: 25,
        photos: [],
      },
      {
        review_id: 147660,
        rating: 5,
        summary: 'Neque repellendus modi eius quaerat ab rerum est animi praesentium.',
        recommend: true,
        response: '"Cumque aspernatur illum velit dolores minus."',
        body: 'Quo tempore est recusandae nobis. Asperiores ut qui dolor. Ducimus modi delectus illum tenetur excepturi.',
        date: '2021-01-10T00:00:00.000Z',
        reviewer_name: 'Celestine_Glover',
        helpfulness: 27,
        photos: [],
      },
      {
        review_id: 147676,
        rating: 4,
        summary: 'Officia reiciendis quaerat nam tempore ut ut numquam ut.',
        recommend: true,
        response: null,
        body: 'Earum ut laborum reprehenderit ipsa tempore sit. Odit rerum cum vitae autem eveniet. Qui qui repellat.',
        date: '2020-12-22T00:00:00.000Z',
        reviewer_name: 'Dameon63',
        helpfulness: 28,
        photos: [],
      }],
    reviews: [{
      review_id: 147660,
      rating: 5,
      summary: 'Neque repellendus modi eius quaerat ab rerum est animi praesentium.',
      recommend: true,
      response: '"Cumque aspernatur illum velit dolores minus."',
      body: 'Quo tempore est recusandae nobis. Asperiores ut qui dolor. Ducimus modi delectus illum tenetur excepturi.',
      date: '2021-01-10T00:00:00.000Z',
      reviewer_name: 'Celestine_Glover',
      helpfulness: 27,
      photos: [],
    },
    {
      review_id: 147676,
      rating: 4,
      summary: 'Officia reiciendis quaerat nam tempore ut ut numquam ut.',
      recommend: true,
      response: null,
      body: 'Earum ut laborum reprehenderit ipsa tempore sit. Odit rerum cum vitae autem eveniet. Qui qui repellat.',
      date: '2020-12-22T00:00:00.000Z',
      reviewer_name: 'Dameon63',
      helpfulness: 28,
      photos: [],
    }],
    loadMoreReviews: () => {
      const loadArr = [];
      let count = 0;
      let totalLength = loadArr.length + this.state.reviews.length;
      while (count < this.state.displayLimit && totalLength < this.state.fullreviewsArr.length) {
        loadArr.push(this.state.fullreviewsArr[totalLength]);
        count += 1;
        totalLength += 1;
      }
      this.setState((prevState) => ({
        reviews: [...prevState.reviews, ...loadArr],
      }));
    }
  };

  let ReviewsListComponent, ReviewsComponent;
  beforeEach(() => {
    ReviewsListComponent = mount(<ReviewsList {...props} debug />);
    // ReviewsComponent = mount(<Reviews debug />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


  it('Should render correctly', () => {
    expect(ReviewsListComponent).toMatchSnapshot();
  });

  it('Should render 2 review tiles at a time', () => {
    expect(ReviewsListComponent.find(ReviewTile).length).toEqual(2);
  });

  // it('Should render 2 more review tiles by clicking MORE REVIEW button', () => {
  //   const spyOnClick = jest.spyOn(ReviewsComponent.instance(), 'loadMoreReviews');
  //   // ReviewsListComponent.instance().loadMoreReviews();
  //   const moreReviews = ReviewsListComponent.find('button').at(0);
  //   moreReviews.instance().loadMoreReviews();
  //   expect(spyOnClick).toHaveBeenCalled();
  //   expect(ReviewsListComponent.find(ReviewTile).length).toEqual(4);
  // });
});
