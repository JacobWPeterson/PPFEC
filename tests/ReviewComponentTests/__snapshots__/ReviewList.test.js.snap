// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList Component Should render correctly 1`] = `
<ReviewsList
  debug={true}
  fullreviewsArr={
    Array [
      Object {
        "body": "Non amet dolorem ut tempora perferendis sint ut adipisci suscipit. Sint ea itaque debitis quos enim odit. Ad perferendis nihil quisquam est. Laboriosam fuga ea amet repellat. Blanditiis aspernatur nostrum aspernatur quas necessitatibus.",
        "date": "2021-01-26T00:00:00.000Z",
        "helpfulness": 12,
        "photos": Array [],
        "rating": 2,
        "recommend": true,
        "response": null,
        "review_id": 147659,
        "reviewer_name": "Shyanne6",
        "summary": "Tenetur quo blanditiis necessitatibus eveniet quo ipsa quia alias voluptas.",
      },
      Object {
        "body": "Corporis voluptate dolore earum voluptas numquam. Perspiciatis cumque et. Dignissimos asperiores ab repudiandae. Consectetur et exercitationem hic. Ut repudiandae dolorum id.",
        "date": "2021-01-20T00:00:00.000Z",
        "helpfulness": 25,
        "photos": Array [],
        "rating": 1,
        "recommend": false,
        "response": null,
        "review_id": 147667,
        "reviewer_name": "Janiya_Frami98",
        "summary": "Minima repudiandae eligendi nemo sit quia aspernatur.",
      },
      Object {
        "body": "Quo tempore est recusandae nobis. Asperiores ut qui dolor. Ducimus modi delectus illum tenetur excepturi.",
        "date": "2021-01-10T00:00:00.000Z",
        "helpfulness": 27,
        "photos": Array [],
        "rating": 5,
        "recommend": true,
        "response": "\\"Cumque aspernatur illum velit dolores minus.\\"",
        "review_id": 147660,
        "reviewer_name": "Celestine_Glover",
        "summary": "Neque repellendus modi eius quaerat ab rerum est animi praesentium.",
      },
      Object {
        "body": "Earum ut laborum reprehenderit ipsa tempore sit. Odit rerum cum vitae autem eveniet. Qui qui repellat.",
        "date": "2020-12-22T00:00:00.000Z",
        "helpfulness": 28,
        "photos": Array [],
        "rating": 4,
        "recommend": true,
        "response": null,
        "review_id": 147676,
        "reviewer_name": "Dameon63",
        "summary": "Officia reiciendis quaerat nam tempore ut ut numquam ut.",
      },
    ]
  }
  reviews={
    Array [
      Object {
        "body": "Quo tempore est recusandae nobis. Asperiores ut qui dolor. Ducimus modi delectus illum tenetur excepturi.",
        "date": "2021-01-10T00:00:00.000Z",
        "helpfulness": 27,
        "photos": Array [],
        "rating": 5,
        "recommend": true,
        "response": "\\"Cumque aspernatur illum velit dolores minus.\\"",
        "review_id": 147660,
        "reviewer_name": "Celestine_Glover",
        "summary": "Neque repellendus modi eius quaerat ab rerum est animi praesentium.",
      },
      Object {
        "body": "Earum ut laborum reprehenderit ipsa tempore sit. Odit rerum cum vitae autem eveniet. Qui qui repellat.",
        "date": "2020-12-22T00:00:00.000Z",
        "helpfulness": 28,
        "photos": Array [],
        "rating": 4,
        "recommend": true,
        "response": null,
        "review_id": 147676,
        "reviewer_name": "Dameon63",
        "summary": "Officia reiciendis quaerat nam tempore ut ut numquam ut.",
      },
    ]
  }
  reviewsMeta={
    Object {
      "characteristics": Object {
        "Comfort": Object {
          "id": 50011,
          "value": "2.8636363636363636",
        },
        "Fit": Object {
          "id": 50009,
          "value": "3.0454545454545455",
        },
        "Length": Object {
          "id": 50010,
          "value": "3.2727272727272727",
        },
        "Quality": Object {
          "id": 50012,
          "value": "3.1818181818181818",
        },
      },
      "product_id": "14930",
      "ratings": Object {
        "1": "5",
        "2": "7",
        "3": "3",
        "4": "5",
        "5": "2",
      },
      "recommended": Object {
        "false": "5",
        "true": "17",
      },
    }
  }
>
  <ReviewsList__ReviewsWrapper>
    <div
      className="ReviewsList__ReviewsWrapper-sc-1ulmjy1-0 eNhjVC"
    >
      <ReviewsList__ReviewSortWrapper>
        <div
          className="ReviewsList__ReviewSortWrapper-sc-1ulmjy1-4 ctmMjX"
        >
          <h2>
            22
             
            reviews, sorted by
          </h2>
          <ReviewsList__SelectTag
            onChange={[Function]}
          >
            <select
              className="ReviewsList__SelectTag-sc-1ulmjy1-5 gUTFzi"
              onChange={[Function]}
            >
              <option
                defaultValue="relevant"
              >
                Relevant
              </option>
              <option
                value="helpful"
              >
                Helpful
              </option>
              <option
                value="newest"
              >
                Newest
              </option>
            </select>
          </ReviewsList__SelectTag>
        </div>
      </ReviewsList__ReviewSortWrapper>
      <ReviewsList__ListWrapper>
        <div
          className="ReviewsList__ListWrapper-sc-1ulmjy1-2 qQXjV"
        >
          <ReviewTile
            key="147660"
            review={
              Object {
                "body": "Quo tempore est recusandae nobis. Asperiores ut qui dolor. Ducimus modi delectus illum tenetur excepturi.",
                "date": "2021-01-10T00:00:00.000Z",
                "helpfulness": 27,
                "photos": Array [],
                "rating": 5,
                "recommend": true,
                "response": "\\"Cumque aspernatur illum velit dolores minus.\\"",
                "review_id": 147660,
                "reviewer_name": "Celestine_Glover",
                "summary": "Neque repellendus modi eius quaerat ab rerum est animi praesentium.",
              }
            }
          >
            <ReviewTile__TileContainer
              reportClick={false}
            >
              <div
                className="ReviewTile__TileContainer-sc-1vbuo4p-0 izQjHl"
              >
                <ReviewTile__StarDateWrapper>
                  <div
                    className="ReviewTile__StarDateWrapper-sc-1vbuo4p-1 darHKr"
                  >
                    <RatingStars
                      rating={5}
                    >
                      <div>
                        <RatingStars__StarsContainer>
                          <div
                            className="RatingStars__StarsContainer-sc-1djn8fm-0 cYnGWG"
                          >
                            <RatingStars__StarsContainer
                              className="top"
                              rating={5}
                            >
                              <div
                                className="RatingStars__StarsContainer-sc-1djn8fm-0 VrplK top"
                              >
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                              </div>
                            </RatingStars__StarsContainer>
                            <RatingStars__StarsContainer
                              className="bottom"
                            >
                              <div
                                className="RatingStars__StarsContainer-sc-1djn8fm-0 cYnGWG bottom"
                              >
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                              </div>
                            </RatingStars__StarsContainer>
                          </div>
                        </RatingStars__StarsContainer>
                      </div>
                    </RatingStars>
                    <span>
                      Celestine_Glover
                      ,
                       
                      January
                       
                      10
                      ,
                       
                      2021
                    </span>
                  </div>
                </ReviewTile__StarDateWrapper>
                <h3>
                  Neque repellendus modi eius quaerat ab rerum est animi praesentium.
                </h3>
                <p>
                  Quo tempore est recusandae nobis. Asperiores ut qui dolor. Ducimus modi delectus illum tenetur excepturi.
                </p>
                <div>
                  ✓ I recommend this product
                </div>
                <ReviewTile__ResponseWrapper>
                  <div
                    className="ReviewTile__ResponseWrapper-sc-1vbuo4p-2 iCAzHl"
                  >
                    <h4>
                      Response from seller
                    </h4>
                    <p>
                      "Cumque aspernatur illum velit dolores minus."
                    </p>
                  </div>
                </ReviewTile__ResponseWrapper>
                <ReviewTile__HelpfulnessWrapper>
                  <div
                    className="ReviewTile__HelpfulnessWrapper-sc-1vbuo4p-4 doSZNz"
                  >
                    Helpful?
                    <ReviewTile__ClickTag
                      className="not-click"
                      onClick={[Function]}
                    >
                      <div
                        className="ReviewTile__ClickTag-sc-1vbuo4p-5 jSpjYR not-click"
                        onClick={[Function]}
                      >
                        Yes
                         
                        (
                        27
                        )
                      </div>
                    </ReviewTile__ClickTag>
                    <div>
                        |  
                    </div>
                    <ReviewTile__ClickTag
                      className="not-click"
                      onClick={[Function]}
                    >
                      <div
                        className="ReviewTile__ClickTag-sc-1vbuo4p-5 jSpjYR not-click"
                        onClick={[Function]}
                      >
                        Report
                      </div>
                    </ReviewTile__ClickTag>
                  </div>
                </ReviewTile__HelpfulnessWrapper>
              </div>
            </ReviewTile__TileContainer>
          </ReviewTile>
          <ReviewTile
            key="147676"
            review={
              Object {
                "body": "Earum ut laborum reprehenderit ipsa tempore sit. Odit rerum cum vitae autem eveniet. Qui qui repellat.",
                "date": "2020-12-22T00:00:00.000Z",
                "helpfulness": 28,
                "photos": Array [],
                "rating": 4,
                "recommend": true,
                "response": null,
                "review_id": 147676,
                "reviewer_name": "Dameon63",
                "summary": "Officia reiciendis quaerat nam tempore ut ut numquam ut.",
              }
            }
          >
            <ReviewTile__TileContainer
              reportClick={false}
            >
              <div
                className="ReviewTile__TileContainer-sc-1vbuo4p-0 izQjHl"
              >
                <ReviewTile__StarDateWrapper>
                  <div
                    className="ReviewTile__StarDateWrapper-sc-1vbuo4p-1 darHKr"
                  >
                    <RatingStars
                      rating={4}
                    >
                      <div>
                        <RatingStars__StarsContainer>
                          <div
                            className="RatingStars__StarsContainer-sc-1djn8fm-0 cYnGWG"
                          >
                            <RatingStars__StarsContainer
                              className="top"
                              rating={4}
                            >
                              <div
                                className="RatingStars__StarsContainer-sc-1djn8fm-0 jmELFT top"
                              >
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                              </div>
                            </RatingStars__StarsContainer>
                            <RatingStars__StarsContainer
                              className="bottom"
                            >
                              <div
                                className="RatingStars__StarsContainer-sc-1djn8fm-0 cYnGWG bottom"
                              >
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                                <span>
                                  ★
                                </span>
                              </div>
                            </RatingStars__StarsContainer>
                          </div>
                        </RatingStars__StarsContainer>
                      </div>
                    </RatingStars>
                    <span>
                      Dameon63
                      ,
                       
                      December
                       
                      22
                      ,
                       
                      2020
                    </span>
                  </div>
                </ReviewTile__StarDateWrapper>
                <h3>
                  Officia reiciendis quaerat nam tempore ut ut numquam ut.
                </h3>
                <p>
                  Earum ut laborum reprehenderit ipsa tempore sit. Odit rerum cum vitae autem eveniet. Qui qui repellat.
                </p>
                <div>
                  ✓ I recommend this product
                </div>
                <ReviewTile__HelpfulnessWrapper>
                  <div
                    className="ReviewTile__HelpfulnessWrapper-sc-1vbuo4p-4 doSZNz"
                  >
                    Helpful?
                    <ReviewTile__ClickTag
                      className="not-click"
                      onClick={[Function]}
                    >
                      <div
                        className="ReviewTile__ClickTag-sc-1vbuo4p-5 jSpjYR not-click"
                        onClick={[Function]}
                      >
                        Yes
                         
                        (
                        28
                        )
                      </div>
                    </ReviewTile__ClickTag>
                    <div>
                        |  
                    </div>
                    <ReviewTile__ClickTag
                      className="not-click"
                      onClick={[Function]}
                    >
                      <div
                        className="ReviewTile__ClickTag-sc-1vbuo4p-5 jSpjYR not-click"
                        onClick={[Function]}
                      >
                        Report
                      </div>
                    </ReviewTile__ClickTag>
                  </div>
                </ReviewTile__HelpfulnessWrapper>
              </div>
            </ReviewTile__TileContainer>
          </ReviewTile>
        </div>
      </ReviewsList__ListWrapper>
      <ReviewsList__ButtonWrapper>
        <div
          className="ReviewsList__ButtonWrapper-sc-1ulmjy1-1 gExWQo"
        >
          <ReviewsList__ReviewButton>
            <button
              className="ReviewsList__ReviewButton-sc-1ulmjy1-3 jjgJVM"
            >
              MORE REVIEWS
            </button>
          </ReviewsList__ReviewButton>
          <ReviewsList__ReviewButton>
            <button
              className="ReviewsList__ReviewButton-sc-1ulmjy1-3 jjgJVM"
            >
              ADD A REVIEW
            </button>
          </ReviewsList__ReviewButton>
        </div>
      </ReviewsList__ButtonWrapper>
    </div>
  </ReviewsList__ReviewsWrapper>
</ReviewsList>
`;
